File: LeetCode/Simulation/Golang/Leetcode1184-distance-between-bus-stops.go
package sbgo

func distanceBetweenBusStops(distance []int, start int, destination int) int {
	if start > destination {
		start, destination = destination, start
	}
	s1, s2 := 0, 0
	for i := 0; i < len(distance); i++ {
		if i >= start && i < destination {
			s1 += distance[i]
		} else {
			s2 += distance[i]
		}
	}
	return min(s1, s2)
}

func min(i, j int) int {
	if i < j {
		return i
	}
	return j
}

File: LeetCode/Simulation/Rust/Leetcode1184-distance-between-bus-stops.rs
impl Solution {
  pub fn distance_between_bus_stops(distance: Vec<i32>, mut start: i32, mut destination: i32) -> i32 {
    if start > destination {
      std::mem::swap(&mut start, &mut destination)
    }
    let (start, destination) = (start as usize, destination as usize);
    distance[start..destination]
      .iter()
      .sum::<i32>()
      .min(distance[0..start].iter().sum::<i32>() + distance[destination..].iter().sum::<i32>())
  }
}
